botocore/awsrequest.py

```diff
@@ -534,6 +557,84 @@ def reset_stream(self):
             raise UnseekableStreamError(stream_object=self.body)
 
 
+class AWSRequestCompressor:
+    """A class that can compress the body of an ``AWSRequest``."""
+
+    def compress(self, body, encodings, headers):
+        """Compresses the reqeust body using the specified encodings.
+
+        Set or append the `Content-Encoding` header with the matched encoding
+        if not present.
+        """
+        for encoding in encodings:
+            encoder = SUPPORTED_COMPRESSION_ENCODINGS.get(encoding)
+            if encoder is not None:
+                if 'Content-Encoding' not in headers:
+                    headers['Content-Encoding'] = encoding
+                elif encoding not in headers['Content-Encoding']:
+                    headers.replace_header(
+                        'Content-Encoding',
+                        f'{headers["Content-Encoding"]},{encoding}',
+                    )
+                return self._compress_body(body, encoder)
+            else:
+                logger.debug('Unsupported compression encoding: %s' % encoding)
```

Comment:
Are you suggesting raising an error then? Because that seems awfully risky if a broken model somehow made it through to production and we elect to error out instead of letting requests proceed with uncompressed bodies.